source/common/router/retry_state_impl.cc

```diff
@@ -58,8 +71,17 @@ RetryStateImpl::RetryStateImpl(const RetryPolicy& route_policy,
       retriable_status_codes_(route_policy.retriableStatusCodes()),
       retriable_headers_(route_policy.retriableHeaders()),
       reset_headers_(route_policy.resetHeaders()),
-      reset_max_interval_(route_policy.resetMaxInterval()) {
-
+      reset_max_interval_(route_policy.resetMaxInterval()),
+      conn_pool_new_stream_with_early_data_and_alt_svc_(
+          conn_pool_new_stream_with_early_data_and_alt_svc) {
+  if (conn_pool_new_stream_with_early_data_and_alt_svc_ &&
+      (cluster.features() & Upstream::ClusterInfo::Features::HTTP3) &&
+      Http::Utility::isZeroRttSafeRequest(request_headers)) {
+    // Always retry potential 0-RTT requests if they are rejected.
```

Comment:
Can we make this comment a bit more detail: if they're rejected due to the request being sent too early